### PR TITLE
Added the manager to manager/cloud_admin ticket approval

### PR DESCRIPTION
**Pull request by athiruma**
1. When a manager is on PTO, the user is on waiting for their approval. So we can send mail to the skip manager to approve the request, after 2 days skip manager is not approved. We send mail to cloud-admin to approve the request.
2. Removed the user/manager cc from the manager approval mail and add the tag TicketId.

### CODE REVIEW

File: cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py [hunk 1]
```diff
@@ -81,19 +84,28 @@ def __send_ticket_status_alerts(self, tickets: dict, ticket_status: str):
...
+                        ticket_opened_days = (current_date - ticket_opened_date).days
                         if ticket_status == self.NEW:  # alert manager if didn't take any action
                             to = manager
-                            subject, body = self.__mail_message.cro_request_for_manager_approval(manager=to, request_user=user, cloud_name=self.__cloud_name, ticket_id=ticket_id, description=description)
-                            cc.append(description.get('EmailAddress'))
+                            extra_message = ''
+                            if self.__manager_escalation_days <= ticket_opened_days <= self.__manager_escalation_days+2:
+                                manager_details = self.__ldap_search.get_user_details(user_name=manager)
+                                to = manager_details.get('ManagerId')
+                                extra_message = f"Your associate/Manager: [{manager}] doesn't approve this request. Please review the below details and approve"
```
Comment (athiruma) [inline cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py:101]: i did it here

File: cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py [hunk 2]
```diff
@@ -81,19 +84,28 @@ def __send_ticket_status_alerts(self, tickets: dict, ticket_status: str):
...
                         cc = self.__default_admins
                         subject = body = to = None
+                        ticket_opened_days = (current_date - ticket_opened_date).days
                         if ticket_status == self.NEW:  # alert manager if didn't take any action
                             to = manager
-                            subject, body = self.__mail_message.cro_request_for_manager_approval(manager=to, request_user=user, cloud_name=self.__cloud_name, ticket_id=ticket_id, description=description)
-                            cc.append(description.get('EmailAddress'))
+                            extra_message = ''
+                            if self.__manager_escalation_days <= ticket_opened_days <= self.__manager_escalation_days+2:
+                                manager_details = self.__ldap_search.get_user_details(user_name=manager)
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py:99]: Maybe we should call it top_manager_details

File: cloud_governance/common/mails/mail_message.py [hunk 3]
```diff
@@ -310,7 +310,7 @@ def cro_cost_over_usage(self, **kwargs):
 """
         return subject, body
 
-    def cro_request_for_manager_approval(self, manager: str, request_user: str, cloud_name: str, ticket_id: str, description: dict):
+    def cro_request_for_manager_approval(self, manager: str, request_user: str, cloud_name: str, ticket_id: str, description: dict, **kwargs):
```
Comment (krishvoor) [inline cloud_governance/common/mails/mail_message.py:313]: Good tackle via **kwargs, this will eventually increase in future refined.

**Review by krishvoor — APPROVED** [review]

File: cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py [hunk 4]
```diff
@@ -81,19 +85,29 @@ def __send_ticket_status_alerts(self, tickets: dict, ticket_status: str):
...
-                            cc.append(description.get('EmailAddress'))
+                            extra_message = ''
+                            if self.__manager_escalation_days <= ticket_opened_days <= self.__manager_escalation_days + 2:
+                                manager_of_manager = self.__ldap_search.get_user_details(user_name=manager)
+                                to = manager_of_manager.get('ManagerId', '')
+                                extra_message = f"Your associate/Manager: [{manager}] doesn't approve this request.<br/>The user {user} is waiting for approval for last {ticket_opened_days} days.<br/>" \
+                                                f"Please review the below details and approve/ reject"
+                            elif ticket_opened_days >= self.__manager_escalation_days + 2:
+                                to = self.__global_admin_name
+                                extra_message = f"<b>CloudAdmin</b> -  Missing manager approval.<br />The user {user} is waiting for approval for last {ticket_opened_days} days.<br/>Please review the below details and approve/reject"
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py:106]: if you sent it to cloud admin, why you need to mention:
`<b>CloudAdmin</b> -  Missing manager approval `
I think you should write something like that:

```
Hi <Cloud admin name>,
The user <user> is waiting for manager <manager> approval more than 5 days.
Please review below ticket details and approve/reject the ticket.
```

File: cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py [hunk 5]
```diff
@@ -81,19 +85,29 @@ def __send_ticket_status_alerts(self, tickets: dict, ticket_status: str):
...
+                        ticket_opened_days = (current_date - ticket_opened_date).days
                         if ticket_status == self.NEW:  # alert manager if didn't take any action
                             to = manager
-                            subject, body = self.__mail_message.cro_request_for_manager_approval(manager=to, request_user=user, cloud_name=self.__cloud_name, ticket_id=ticket_id, description=description)
-                            cc.append(description.get('EmailAddress'))
+                            extra_message = ''
+                            if self.__manager_escalation_days <= ticket_opened_days <= self.__manager_escalation_days + 2:
+                                manager_of_manager = self.__ldap_search.get_user_details(user_name=manager)
+                                to = manager_of_manager.get('ManagerId', '')
+                                extra_message = f"Your associate/Manager: [{manager}] doesn't approve this request.<br/>The user {user} is waiting for approval for last {ticket_opened_days} days.<br/>" \
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py:102]:
```
Hi <Manager name>,
The user <user> is waiting for manager <manager> approval more than 3 days.
Please review below ticket details and approve/reject the ticket.
```

Comment (athiruma) [reply cloud_governance/cloud_resource_orchestration/aws/ec2/monitor_tickets.py:103]: The above message is Good. Do I need to change as you wrote

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**